include/mata/nfa/strings.hh

```diff
@@ -285,11 +285,11 @@ private:
 
 /// A noodle is represented as a sequence of segments (a copy of the segment automata) created as if there was exactly
 ///  one ε-transition between each two consecutive segments.
-using Noodle = std::vector<SharedPtrAut>;
+using Noodle = std::vector<std::shared_ptr<Nfa::Nfa>>;
 using NoodleSequence = std::vector<Noodle>; ///< A sequence of noodles.
 
 /// Noodles as segments enriched with EpsCntMap
-using NoodleSubst = std::vector<std::pair<SharedPtrAut, EpsCntVector>>;
```

Comment:
Are you truly happy with the `EpsCuntVector`?

Reply:
I renamed the types related to this to more desriptive names.

Reply:
I would maybe remove the name NoodleSequence, use just std::vector<Noodle>,
and invent name X for std::pair<SharedPtrAut, EpsCntVector>, 
and replace NoodleSubst by std::vector<X>. 
The NoodleSubst is quite criptic.

Reply:
I will think of something.

Reply:
Refactored. The changes will appear in next PR.